### DISCUSSION TITLE
How to use Capacitor x TypeORM x SQLite x Quasar?

### DISCUSSION THREAD

**Discussion by Ayoskilolo** [capacitor-community/sqlite#401] [Q&A]
I'm trying to make an app that uses Capacitor Sqlite as a lightweight db (an android app for now). I'm building this app using the quasar framework and capacitor. To manage the DB, I'd like to use TypeORM. So far, I've been using the vue-typeorm-app as a guide and this is what I've been able to accomplish:

1. Set up the boot file in quasar to start the DB.
2. Set up the Data source using capacitor community sqlite's driver as a connection.
3. Created my entities. 

Now I'm getting an error in the terminal that says Column TypeUndefined Error.

`boot error: ColumnTypeUndefinedError: Column type for User#firstName is not defined and cannot be guessed. Make sure you have turned on an "emitDecoratorMetadata": true option in tsconfig.json. Also make sure you have imported "reflect-metadata" on top of the main entry file in your application (before any entity imported).If you are using JavaScript instead of TypeScript you must explicitly provide a column type.
 `
 
 I think this is because I've not been able to run a migration that builds the table with SQL. I've tried generating a migration to create this table but I get an error:
  
`Error: The jeep-sqlite element is not present in the DOM! Please check the @capacitor-community/sqlite documentation for instructions regarding the web platform.`

I understand the error because of what I saw in the vue-typeorm-app repo, but I don't know how to implement this in Quasar. My problem is using jeep-sqlite in quasar's weird way of setting up main.js. Could anyone point me as to how?

Lastly, does the android use of this package depend on jeep-sqlite as well?

**Comment by jepiqueau** [comment]
@Ayoskilolo i do not know quasar. What i could suggest is 
Make it working in pure Vue first followin the vue-typeorm-app and when it works add the Quasar framework.
The migration were created by hand and must be inline wirh your entities. If you want a pure android code you do not need to use jeep-sqlite so uou can remove all the code regarding the Web platform. Hope this will help you. I know some developers use capacitor-community/sqlite with vue, typeorm and Quasar. So you should be able to use it

**Comment by jepiqueau** [comment]
@Ayoskilolo see [Quasar-SQLite-CRUD-App](https://jepiqueau.github.io/2023/11/28/Quasar-SQLite-CRUD-App.html).
